aiohttp_pydantic/oas/view.py

```diff
@@ -170,12 +177,18 @@ def generate_oas(apps: List[Application]) -> dict:
     return oas.spec
 
 
-async def get_oas(request):
+def get_oas(version_spec, title_spec):
     """
-    View to generate the Open Api Specification from PydanticView in application.
+    Wrapper over get_oas for filling spec info
     """
-    apps = request.app["apps to expose"]
-    return json_response(generate_oas(apps))
+    async def get_oas_wrapper(request):
+        """
+        View to generate the Open Api Specification from PydanticView in application.
+        """
+        apps = request.app["apps to expose"]
```

Comment:
You don't need to make a closure, you can get `version_spec` and `title_spec` using 
`request.app["version spec"]` and `request.app["title spec"]`